lib/pipelineTemplateVersionFactory.js

```diff
@@ -149,19 +150,22 @@ class PipelineTemplateVersionFactory extends BaseFactory {
             config.params = {};
         }
 
-        if (!config.params.name || !config.params.namespace) {
+        if (config.params.templateId) {
+            return super.list(config);
+        }
```

Comment:
When `templateId` params exist, why are we not considering `name` and `namespace`?
When `templateId`, `name` and `namespace` are specified, we should return versions associated with the template matching `templateId`, `name` and `namespace`.